tests/ndonnx/test_dtypes.py

```diff
@@ -22,28 +22,51 @@ class Unsigned96Impl(OperationsBlock):
     def equal(self, x, y):
         return custom_equal(x, y)
 
+    def add(self, x, y) -> Array:
+        if x.dtype == Unsigned96() and y.dtype == Unsigned96():
+            max_u32 = ndx.iinfo(ndx.uint32).max
+            overflow_amount = (
+                x.lower.astype(ndx.uint64) + y.lower.astype(ndx.uint64)
+            ) >> 32
+            will_overflow = overflow_amount > 0
+            # No where implemented on unsigned types.
+            lower = ((x.lower + y.lower) * (1 - will_overflow)) + (
+                max_u32 * will_overflow
+            )
+            upper = x.upper + y.upper + overflow_amount
+            return Array._from_fields(
+                Unsigned96(),
+                upper=upper,
+                lower=lower,
+            )
+        elif x.dtype == ndx.uint32 and y.dtype == Unsigned96():
+            return y + x
+        elif x.dtype == Unsigned96() and y.dtype == ndx.uint32:
+            return x + y.astype(Unsigned96())
+        return NotImplemented
+
 
 class Unsigned96(StructType, CastMixin):
     def _fields(self) -> dict[str, StructType | CoreType]:
         return {
-            "upper": ndx.uint32,
```

Comment:
Is there some deeper context that I'm missing that made this change necessary or why is one ordering preferable to the other?

Reply:
No deep meaning - it just makes it possible to get the carry without overflowing for the addition implementation for the tests here.